rc/tools/man.kak

```diff
@@ -64,5 +77,71 @@ The page can be a word, or a word directly followed by a section number between
             ;;
     esac
 
-    printf %s\\n "evaluate-commands -try-client %opt{docsclient} man-impl *man* $pagenum $subject"
+    printf %s\\n "evaluate-commands -try-client %opt{docsclient} man-impl man $pagenum $subject"
 } }
+
+
+
+# The following section of code enables a user
+# to go to next or previous man page links and to follow man page links,
+# for example, apropos(1), that would normally appear in SEE ALSO sections.
+# The user would position the cursor on any character of the link
+# and then press <ret> to change to a buffer showing the man page.
+
+# Regex pattern defining a man page link.
+# Used for determining if a selection, which may just be a link, is a link.
+declare-option -hidden regex man_link1 \
+  [\w_.:-]+\(\d[a-z]*\)
+
+# Same as above but with lookbehind and lookahead patterns.
+# Used for searching for a man page link.
+declare-option -hidden regex man_link2 \
+  "(?:^|(?<=\W))%opt{man_link1}(?=\W)"
+
+# Define a useful command sequence for searching a given regex
+# and a given sequence of search keys.
+define-command man-search -params 2 %{
+  set-register / %arg[1]
+  execute-keys %arg[2]
+}
+
+define-command -docstring 'Go to next man page link' \
+man-link-next %{ man-search %opt[man_link2] n }
```

Comment:
I am a bit sceptical that we need all those features in man.kak. Does anybody really iterates on links inside a man page ? I would expect that to be extremely rare, and users only to follow links they voluntarily select.

Reply:
Some man pages list a lot of useful man page links in their SEE ALSO section that one would like to follow. `man-link-next` is convenient for quickly navigating to the link one wants to focus on, because the command preloads register / and then one can use `n` or `<alt-n>`. Otherwise, one needs to use arrow keys or do an ad-hoc search.

In my recent force-push, I've made `man-link-here` a hidden command and omitted it from the suggested key maps. It was mainly exposed for testing, but it's probably not directly useful.